azerothcore/azerothcore-wotlk
# fix(Core/SmartScripts): LOS check  spell  for SmartScipts

## Changes Proposed:

This PR proposes changes to:

## Issues Addressed:
After merging https://github.com/azerothcore/azerothcore-wotlk/pull/18026, we faced with some regressions in SAI scripts. This PR should address issue https://github.com/azerothcore/azerothcore-wotlk/issues/18181 and similar cases. The issue described in https://github.com/azerothcore/azerothcore-wotlk/issues/18181 occurs because in SAI, we attempt to cast https://www.wowhead.com/wotlk/spell=46310/drop-war-halberd by https://www.wowhead.com/wotlk/npc=25802/kaw-the-mammoth-destroyer. However, due to the changes in https://github.com/azerothcore/azerothcore-wotlk/pull/18026, the cast never occurs because those changes do not take into account that the spell can ignore LOS, as in our case.

- Closes  https://github.com/azerothcore/azerothcore-wotlk/issues/18181 

## SOURCE:

The changes have been validated through:

## Tests Performed:

This PR has been:


## How to Test the Changes:



1.
2.
3.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

thx @walkline for help
